Dynmap Marker Functionality

OshDubh opened the pull request:
A discord slash command interface with the assorted dynmap marker commands, allowing people to manage dynmap point markers, upload new marker icons, create marker sets and regions from within discord.

OshDubh committed:
templating addmarker

laying out the initial functions to interact with the basic dynmap
addmarker command

OshDubh committed:
added marker world option choice

OshDubh committed:
registry of addmarker command using retrieved choice fields

OshDubh committed:
mock minecraft exec responses with regex parsing

OshDubh committed:
parsing of addmarker command into marker struct

OshDubh committed:
replaces mock data with the real executed minecraft command

OshDubh committed:
better error handling

OshDubh committed:
removed mocks, cleaning up

smallblue2 commented:
Code looks great! Thanks for putting in the effort for this, will try it out now.

OshDubh committed:
abstracts parsing for testing, adds test representing actual Exec output

OshDubh committed:
tests for ParseMarkerIcons and ParseMarkerSets

OshDubh committed:
correct regex for actual input

OshDubh committed:
marker set test without optional persistent field

OshDubh committed:
extends handler registry to allow for commands to register autocomplete

OshDubh committed:
removes options for icon choice, adds autocompleting options instead

discord does not allow for more than 25 choices to be registered as part
of slash command options, so this change replaces the prepopulated list
of choices with a dynamic list of autocompleting options

OshDubh committed:
Merge branch 'smallblue2:main' into dynmap-markers

OshDubh committed:
sets and worlds switch to autocomplete if we have too many choices
